doc/sflow/sflow_hld.md

```diff
@@ -0,0 +1,459 @@
+* show runningconfiguration sflow
+  * Displays the current running configuration of sflow
+
+### 6.5 **DB and Schema changes**
+
+#### ConfigDB Table & Schema
+
+A new SFLOW_GENETLINK ConfigDB table entry would be added:
+```
+; Defines schema for SFLOW_GENETLINK table which has the genetlink family and multicast group IDs to be used as Host Interfaces
+key                                   = SFLOW_GENETLINK:genetlink_name
+; field                               = value
+genetlink_name                        = 1*16VCHAR                        ; name of the genetlink family
+genetlink_group                       = 1*4DIGIT                         ; genetlink multicast group id
+```
+
+A new SFLOW_COLLECTOR ConfigDB table entry would be added. 
+```
+SFLOW_COLLECTOR|{{collector_name}}
+    "collector_ip": {{ip_address}}
+    "agent_addr": {{ip_address}}
+    "collector_port": {{ uint32 }} (OPTIONAL)
+    "max_datagram_size": {{ uint32 }} (OPTIONAL)
+
+; Defines schema for sFlow collector configuration attributes
+key                                   = SFLOW_COLLECTOR:collector_name   ; sFlow collector configuration
+; field                               = value
+collector_ip                          = IPv4address / IPv6address        ; Ipv4 or IpV6 collector address
+agent_ip                              = IPv4address / IPv6address        ; Ipv4 or IpV6 agent address
+collector_port                        = 1*5DIGIT                         ; destination L4 port : a number between 0 and 65535
+max_datagram_size                     = 1*4DIGIT                         ; MTU of the sflow datagram
+
+;value annotations
+collector_name                        = 1*16VCHAR
+```
+
+A new SFLOW_CONFIG table would be added.
+```
+; Defines schema for SFLOW_CONFIG table which holds global configurations
+key                                   = SFLOW_CONFIG:Config
+; field                               = value
+sampling_rate       = 1*5DIGIT      ; 1 / (sflow sampling rate as a percentage of bandwidth)
+```
+
+#### AppDB & Schema
+
+A new TRAP_GENETLINK table entry would be added:
+
+```
+; Defines schema for TRAP_GENETLINK table which maps trap types that need to be punted to a genetlink channel
+key                                   = TRAP_GENETLINK:trap_id             ; TRAP ID
+; field                               = value
+genetlink_name                        = 1*16VCHAR                          ; name of the genetlink family
+genetlink_group                       = 1*4DIGIT                           ; genetlink multicast group id
+```
+
+A new SFLOW_TABLE is added to the AppDB:
+
+```
+; Defines schema for SFLOW_TABLE table which contains parameters that need to be set for sFlow
+key                 = SFLOW_TABLE:sflow
+; field             = value
+sampling_rate       = 1*5DIGIT           ; 1 / (sflow sampling rate as a percentage of bandwidth)

```

Comment:
this is unclear to me. what does 1 mean here? does it mean we sample every packet?

Reply:
Update definition of sampling rate (it is as per the SAI definition).